src/components/ClaimENSPage/ClaimENSPage.tsx

```diff
@@ -76,14 +77,24 @@ export default class ClaimENSPage extends React.PureComponent<Props, State> {
     this.setState({ isLoading: false })
   }
 
-  handleClaim = () => {
+  handleClaim = async () => {
     const { onOpenModal } = this.props
     const { name } = this.state
-    onOpenModal('ClaimNameFatFingerModal', { originalName: name })
+    const isRepeated = !(await isNameAvailable(name))
```

Comment:
This way it is working but Imo I feel easier to read it as follows:
I would use `const isNameAvailable = await isNameAvailable(name)`

Thoughts @nicosantangelo ?

Reply:
you can't really repeat the name (it'll override the import), but maybe `const isAvailable = await isNameAvailable()` ?

Reply:
Yes, I meant to avoid `!(await) promise` pattern